### PR TITLE
Dont store transition val

### LINKED ISSUE #26866: [BUG] don't store intermediate value into kvs when transitioning
### Reproduction steps

- run some command like `move-to-level` with a non-zero transition time
- the intermediate values will be persisted in the kvs
- this can be undesirable for the following reasons
  - writing to kvs (typically an external flash on many platforms) can be time consuming and will affect the timing accuracy
  - reduce the lifetime of the flash significantly, for eg moving from 1 to 254 will require 253 writes into the flash

### Bug prevalence

always

### GitHub hash of the SDK that was being used

ad5253ace34269d07ca8afbd984aec319903ac56

### Platform

other

### Platform Version(s)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by step0035**
- fix #26866 
- for attributes that supports transitioning, added a `isTemp` argument in their `Set` function
- if the transition is not completed yet, the attribute value is temporary and don't store it into kvs
- only store to kvs when the transition is completed
- for all other attributes, set `isTemp` to false

### Note
- i only added this for levelcontrol and colorcontrol attributes, not sure about other clusters, please help to add it in if i missed out any

### CODE REVIEW

**Comment by step0035** [comment]
will it be better if i create a separate `set` function for the affected attribute? Seems like there are a lot of build failures.

**Comment by Damian-Nordic** [comment]
Note that we solved this issue using `DeferredAttributePersister` not to make changes in the attribute storage: https://github.com/project-chip/connectedhomeip/pull/23366. But maybe your approach is more straightforward, indeed.

File: src/app/util/attribute-table.cpp [hunk 1]
```diff
@@ -54,18 +54,19 @@ EmberAfStatus emberAfWriteAttributeExternal(EndpointId endpoint, ClusterId clust
...
     case EmberAfAttributeWritePermission::AllowWriteOfReadOnly:
         return emAfWriteAttribute(endpoint, cluster, attributeID, dataPtr, dataType,
-                                  (extWritePermission == EmberAfAttributeWritePermission::AllowWriteOfReadOnly), false);
+                                  (extWritePermission == EmberAfAttributeWritePermission::AllowWriteOfReadOnly), false, false);
     default:
         return (EmberAfStatus) extWritePermission;
     }
 }
 
 EmberAfStatus emberAfWriteAttribute(EndpointId endpoint, ClusterId cluster, AttributeId attributeID, uint8_t * dataPtr,
```
Comment (tcarmelveilleux) [inline src/app/util/attribute-table.cpp:63]: Suggest using a new method `emberAfWriteAttributeSkipNonVolatile` to be much more explicit, which calls the version with isTemp.

This way, almost none of the accessors below need to be changed.

File: zzz_generated/app-common/app-common/zap-generated/attributes/Accessors.cpp
Comment (tcarmelveilleux) [inline zzz_generated/app-common/app-common/zap-generated/attributes/Accessors.cpp:1]: The template for codegen has to be changed. You cannot manually change files in zzz_generated. See comment higher up (`// THIS FILE IS GENERATED BY ZAP`)

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
This is a spec violation if done in this blanket way.  See https://github.com/project-chip/connectedhomeip/blob/master/src/app/DeferredAttributePersistenceProvider.h for how this should be handled.

**Comment by step0035** [comment]
Thanks for the reviews, i will close this PR and look into `DeferredAttributePersistenceProvider`.

**Pull request closed without merging**